### PR TITLE
Support juju 3.4

### PR DESCRIPTION
**Pull request by rgildein**
Using master branch for zaza with TEST_JUJU3=1 [1] and dropping
bootstack-actions for func in CI to be able to use specific lxd
channel.

Changing to use of Juju 3.4 controller for testing instead of Juju 3.1, using TEST_JUJU3=1 [1] for zaza to install proper libjuju version. 

Simply sanitation of project and add workflow concurrency to avoid running multiple workflows on a single PR. Like this if new commit arrived the previous run will be canceled by new run.

~~blocked until [#655](https://github.com/openstack-charmers/zaza/pull/655) is not merged~~

---
[1]:
https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6

### CODE REVIEW

**Review by aieri — APPROVED** [review]

**Review by Pjack — APPROVED** [review]

File: .github/workflows/check.yaml
Comment (Pjack) [inline .github/workflows/check.yaml:29]: I would like to know the reason that why we cannot use worflows/func.yaml@v2 or v3? 
Do we have plan to retire worflows/func.yaml@v2 ?

Comment (Pjack) [inline .github/workflows/check.yaml]: I'm curious about the reason behind discontinuing the use of func.yaml. Is there a specific reason why we can't simply modify the existing one or introduce a new version? What's the major challenges to upgrading the controller? Initially, I had anticipated simply adjusting the 'juju-channel' value to 3.4.

My concern is that we may modify check.yaml again when we eventually upgrade to version 3.5. Minimizing these changes would be preferable.

Therefore, I prefer to know the whole plan and announce to whole team. thanks!

Comment (rgildein) [reply .github/workflows/check.yaml]: It was dou current version of those workflows does not support lxd-channel change and I hit the issue https://github.com/canonical/charmcraft/issues/1640. That issue was fixed.

My other reasons are:
- The reusable workflow for functional become too complex with too many parameters, so they are not readable what actually will be executed and how. If you open the pipeline, you see the lot of skipped steps, and you have no idea what actually was run.
- Any small change required us to change the workflow and then come back to the charm and if sometimes just created a new branch.
- We already have 4 branches, so it's not easy to change it on a single place.
- With potential Terraform use we can do it from one place, without that I would rather go and create 20 PRs with changing one parameter, than change it for 4 branches and then hope it works. 

This is an experiment and we can have discussion before merging any PR like this.

Comment (rgildein) [reply .github/workflows/check.yaml]: As we internally discuss and the issue https://github.com/canonical/charmcraft/issues/1640 was fixed, I'll revert this to using bootstack-actions again.

**Comment by rgildein** [comment]
Converted to draft, to test zaza version.
Using my version [zaza](https://github.com/rgildein/zaza/tree/chore/no-ref/fix-action-object) to test fix for support 2.9 and 3.4 by zaza. Please do not merge.

**Review by jneo8 — DISMISSED** [review]
Please check why is the CI fail.

**Comment by rgildein** [comment]
> Please check why is the CI fail.

It's due `PasswordAuthentication no`, which comes with new image configuration in `60-cloudimg-settings.conf`, and I tried to fix it with `'echo "PasswordAuthentication yes" > 01-test-settings.conf'` instead of `sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config`.
It was working on my local LXD fine, I'll test it again.

**Comment by rgildein** [comment]
I did not use full path, so that was a problem.

**Review by Pjack — APPROVED** [review]

**Review by aieri — APPROVED** [review]

**Pull request merged**
